<launch>

<arg name = "StartRec" default = "false" />
<arg name="joy_config" default="xbox360"/>
<arg name="joy_dev" default="/dev/input/js0"/>
<arg name="config_filepath" default="$(find teleop_twist_gamepad)/config/$(arg joy_config).config.yaml"/>
<arg name="autorepeat_rate" default="0.0"/>
<arg name="cmd_topic" default="/cmd_vel"/>
<arg name="enable_deadman" default="true"/>

<!-- kamera node -->
<node name="usb_cam" pkg="usb_cam" type="usb_cam_node">
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
  </node>
  <node name="image_view" pkg="image_view" type="image_view" respawn="false">
    <remap from="image" to="/usb_cam/image_raw"/>
    <param name="autosize" value="true" />
  </node>

<!-- Detection node -->
<node pkg="line_follower_turtlebot" name="detection" type="detect" cwd="node" output="screen">
</node>

<!-- Robot commands node -->
<node pkg="line_follower_turtlebot" name="Velocity" type="navig" cwd="node" output="screen">
</node>

<node
pkg = "rosbag"
type = "record"
name = "rosbag_record_all"
respawn = "true"
output = "screen"
args = "-a -O BagFile -x /camera/(.*) "
if = "$(arg StartRec)"
/>

<!-- Detection node -->
<node pkg="line_follower_gelb" name="detection_gelb" type="detect_gelb" cwd="node" output="screen">
</node>

<!--Robot commands node -->
<node pkg="line_follower_gelb" name="gelb" type="navig_gelb" cwd="node" output="screen">
</node>

<node
pkg = "rosbag"
type = "record"
name = "rosbag_record_all"
respawn = "true"
output = "screen"
args = "-a -O BagFile -x /camera/(.*) "
if = "$(arg StartRec)"
/>
<!-- joy node -->
<node pkg="joy" type="joy_node" name="joy_node" output="screen">
<param name="dev" value="$(arg joy_dev)" type="string"/>
<param name="default_trig_val" value="true"/>

<param name="deadzone" value="0.3" type="double"/>
<param name="coalesce_interval" value="0.01" type="double"/>
<param name="autorepeat_rate" value="$(arg autorepeat_rate)" type="double"/>
</node>
<!-- joy-controller node -->
<node pkg="teleop_twist_gamepad" name="teleop_twist_gamepad" type="teleop_node" output="screen">
<rosparam command="load" file="$(arg config_filepath)"/>
<param name="joy_config" value="$(arg joy_config)"/>
<param name="cmd_topic" value="$(arg cmd_topic)"/>
<param name="enable_deadman" value="$(arg enable_deadman)"/>
</node>


<!-- Auswahl node -->
<node pkg="line_oder_joy" name="line_oder_joy" type="line_oder_joy">
</node>
</launch>
